Sources/TuistCore/Graph/Graph.swift

```diff
@@ -326,7 +326,15 @@ public class Graph: Graphing {
             .map(productDependencyReference)
 
         references.formUnion(otherTargetFrameworks)
+		
+        /// Xcframeworks
+        let xcframeworks = findAll(targetNode: targetNode, test: { $0 is XCFrameworkNode })
```

Comment:
What about adding a unit tests for this to make sure we are including the reference of the xcframeworks?

Reply:
I believe `xcFrameworks` will be automatically included in line [317](https://github.com/tuist/tuist/pull/701/files#diff-f721e485c29e079eaa1e86576c6fbc7dR317) as such explicitly searching for them is not needed. Sadly this could also cause an xcframework to be embedded when it doesn't need to be (in case of static frameworks/libraries being bundled in that xcframework)

```swift
let precompiledFrameworks = findAll(targetNode: targetNode, test: isDynamicAndLinkable) ....
```

This is capable of picking up xcframeworks provided the xcframework is able to provide an appropriate binary path which I believe the recent changes should help achieve. 

+1 with @pepibumur on adding tests - it will surface / try this out to see it working.

Reply:
> I believe `xcFrameworks` will be automatically included in line [317](https://github.com/tuist/tuist/pull/701/files#diff-f721e485c29e079eaa1e86576c6fbc7dR317) as such explicitly searching for them is not needed. Sadly this could also cause an xcframework to be embedded when it doesn't need to be (in case of static frameworks/libraries being bundled in that xcframework)
> 
> ```swift
> let precompiledFrameworks = findAll(targetNode: targetNode, test: isDynamicAndLinkable) ....
> ```
> 
> This is capable of picking up xcframeworks provided the xcframework is able to provide an appropriate binary path which I believe the recent changes should help achieve.
> 

I've locally tested the check as @kwridan mentioned is not needed, as xcframeworks are getting picked up during precompiledFrameworks lookup. I have removed the redundant lookup now